src/hooks/ParameterConstraintsHooks.js

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { useSolutionParameters } from '../state/hooks/SolutionHooks';
+import { TranslationUtils, ParameterConstraintsUtils } from '../utils';
+import isBefore from 'date-fns/isBefore';
+import isAfter from 'date-fns/isAfter';
+import isSameDay from 'date-fns/isSameDay';
+import { useCallback, useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import format from 'date-fns/format';
+import isValid from 'date-fns/isValid';
+
+export const useParameterConstraint = (parameterData) => {
+  const { t } = useTranslation();
+  const parametersList = useSolutionParameters();
+  const { trigger, watch } = useFormContext();
+  const constraint = ParameterConstraintsUtils.getParameterValidationConstraint(
+    parameterData?.options?.validation,
+    parameterData.varType,
+    parametersList
+  );
+  const constrainingValue = constraint ? watch(constraint?.id) : undefined;
+
+  useEffect(() => {
+    if (constraint) {
+      trigger(parameterData.id);
+    }
+  }, [constraint, parameterData.id, trigger, constrainingValue]);
+  return { t, constraint, constrainingValue };
```

Comment:
It seems that the `t` function is not used in `useParameterConstraint`, do we need to retrieve it and return it here? (or maybe you were anticipating future hooks, to factorize the code between several types of constraints)

Reply:
Yes, I anticipated it for future hooks (numbers, string etc) because I wanted to avoid declaring `const { t } = useTranslation() `in every hook and since all hooks will use `useParameterConstraint`, I included `t` in its returned value :slightly_smiling_face: